src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory_Methods.cs

```diff
@@ -53,37 +53,12 @@ internal IArgument CreateArgumentOperation(ArgumentKind kind, IParameterSymbol p
 
         private IVariableDeclaration CreateVariableDeclarationInternal(BoundLocalDeclaration boundLocalDeclaration, SyntaxNode syntax)
         {
-            IVariableInitializer initializer = null;
-            if (boundLocalDeclaration.InitializerOpt != null)
-            {
-                IOperation initializerValue = Create(boundLocalDeclaration.InitializerOpt);
-                SyntaxNode initializerSyntax = null;
-                bool isImplicit = false;
-                if (syntax is VariableDeclaratorSyntax variableDeclarator)
-                {
-                    initializerSyntax = variableDeclarator.Initializer;
-                }
-                else
-                {
-                    Debug.Fail($"Unexpected syntax kind: {syntax.Kind()}");
-                }
-
-                if (initializerSyntax == null)
-                {
-                    // There is no explicit syntax for the initializer, so we use the initializerValue's syntax and mark the operation as implicit.
-                    initializerSyntax = initializerValue.Syntax;
-                    isImplicit = true;
-                }
-
-                initializer = OperationFactory.CreateVariableInitializer(initializerSyntax, initializerValue, _semanticModel, isImplicit);
-            }
-
-            return OperationFactory.CreateVariableDeclaration(boundLocalDeclaration.LocalSymbol, initializer, _semanticModel, syntax);
+            return OperationFactory.CreateVariableDeclaration(boundLocalDeclaration.LocalSymbol, Create(boundLocalDeclaration.InitializerOpt), _semanticModel, syntax);
         }
 
         private IVariableDeclaration CreateVariableDeclaration(BoundLocal boundLocal)
         {
```

Comment:
>!ReceiverOpt.WasCompilerGenerated [](start = 53, length = 33)

The fact that receiver is compiler generated doesn't mean that all its descendants are. I think we should also check for specific nodes here: BoundThis, and what else is getting translated into IInstanceReferenceExpression.